CONTRIBUTING.md

```diff
@@ -558,17 +562,17 @@ arguments and return values.
 
 ### Code Review
```

Comment:
Shall we move [code review](https://github.com/GenericMappingTools/pygmt/blob/revise-pr-review-process/CONTRIBUTING.md#code-review) subsection to be within [General guidelines](https://github.com/GenericMappingTools/pygmt/blob/revise-pr-review-process/CONTRIBUTING.md#general-guidelines)? I feel they are related to each other directly, but they are separated by so many subsections that present how to revise PyGMT code and write documentation.

Reply:
> Shall we move [code review](https://github.com/GenericMappingTools/pygmt/blob/revise-pr-review-process/CONTRIBUTING.md#code-review) subsection to be within [General guidelines](https://github.com/GenericMappingTools/pygmt/blob/revise-pr-review-process/CONTRIBUTING.md#general-guidelines)? I feel they are related to each other directly, but they are separated by so many subsections that present how to revise PyGMT code and write documentation.

What do you think? Do you think it's worth putting the code review subsection nearer to the top? I do find that the code review section has some good tips which may be counted as general guidelines.

> How about adding how to request reviewers and making a PR draft in this subsection?

Another thing I'm thinking of is whether we edit the welcome bot message at https://github.com/GenericMappingTools/pygmt/blob/v0.3.1/.github/config.yml (hasn't been modified for 2 years since 2019). Maybe there are things people only need to be told once, and we can simplify the CONTRIBUTING.md guidelines significantly. What do you think?

Reply:
> > Shall we move [code review](https://github.com/GenericMappingTools/pygmt/blob/revise-pr-review-process/CONTRIBUTING.md#code-review) subsection to be within [General guidelines](https://github.com/GenericMappingTools/pygmt/blob/revise-pr-review-process/CONTRIBUTING.md#general-guidelines)? I feel they are related to each other directly, but they are separated by so many subsections that present how to revise PyGMT code and write documentation.
> 
> What do you think? Do you think it's worth putting the code review subsection nearer to the top? I do find that the code review section has some good tips which may be counted as general guidelines.

I think **Code Review** is a key part of every PR, right? So I think it's better to make it as general guidelines for both first and old contributors. Currently, the [general guides](https://github.com/GenericMappingTools/pygmt/blob/revise-pr-review-process/CONTRIBUTING.md#general-guidelines) end by
> -  Be willing to accept criticism and work on improving your code; we don't want to break other users' code, so care must be taken not to introduce bugs.
> - Be aware that the pull request review process is not immediate, and is generally proportional to the size of the pull request.

They are a little related to **code review**. 

Besides, the 1st point of **code review** is the same as the 6th point of **General guidelines**. If we move **code review** in this section, they may be merged.

> Write a good and detailed description of what the PR does.

> Describe what your PR changes and why this is a good thing. Be as specific as you can. The PR description is how we keep track of the changes made to the project over time.

---

**Some other thoughts:**

I guess the two rules tell contributors to finish all the things before submitting a PR so that it can be reviewed and accepted soon. We have good [Testing your code](https://github.com/GenericMappingTools/pygmt/blob/revise-pr-review-process/CONTRIBUTING.md#testing-your-code), [Testing plots](https://github.com/GenericMappingTools/pygmt/blob/revise-pr-review-process/CONTRIBUTING.md#testing-plots), [Tutorials](https://github.com/GenericMappingTools/pygmt/blob/revise-pr-review-process/CONTRIBUTING.md#tutorials) and [Gallery](https://github.com/GenericMappingTools/pygmt/blob/revise-pr-review-process/CONTRIBUTING.md#gallery-plots) guides that can be linked.
> Write tests for the code you wrote/modified if needed.
> Include an example of new features in the gallery or tutorials.

This suggestion is actually explained in detail in [Code Style](https://github.com/GenericMappingTools/pygmt/blob/revise-pr-review-process/CONTRIBUTING.md#code-style). We may add a link to it.
> Follow the PEP8 style guide for code and the numpy guide for documentation.

Reply:
I moved **Code Review** to be within **general Guidelines** in 81d6285. Some content needs revisions as suggested in https://github.com/GenericMappingTools/pygmt/pull/1119#discussion_r604635739 if we think this movement is good:

> Currently, the [general guides](https://github.com/GenericMappingTools/pygmt/blob/revise-pr-review-process/CONTRIBUTING.md#general-guidelines) end by
> 
> > * Be willing to accept criticism and work on improving your code; we don't want to break other users' code, so care must be taken not to introduce bugs.
> > * Be aware that the pull request review process is not immediate, and is generally proportional to the size of the pull request.
> 
> They are a little related to **code review**.
> 
> Besides, the 1st point of **code review** is the same as the 6th point of **General guidelines**. If we move **code review** in this section, they may be merged.
> 
> > Write a good and detailed description of what the PR does.
> 
> > Describe what your PR changes and why this is a good thing. Be as specific as you can. The PR description is how we keep track of the changes made to the project over time.